cql/ManageRareAbnormality.cql

```diff
@@ -1038,23 +1189,27 @@ define RecommendationForHistologyResults:
         HistologyInterpretedAsCin2
       ) and
       HasHistologyInterpretedAsUnspecifiedHsilOrCin2BeforeMostRecentBiopsy and
-      RecentlyRespondedYesToFuturePregnancyConcernsQuestion
+      RecentlyRespondedYesToFuturePregnancyConcernsQuestion and
+      (
+        Collate.DateOfLastCervicalPrecancerTreatment is null or 
+        not (Collate.DateOfLastCervicalPrecancerTreatment occurs after Collate.MostRecentCin2Biopsy.date) or
+        Collate.DateOfLastCervicalPrecancerTreatment 7 years or more before Collate.MostRecentCin2Biopsy.date
+      )
     ) then
       {
         short: 'Surveillance or Treatment',
         date: Today(),
         group: 'Managing Histology (I.3)',
         details: {
-          'Treatment is recommended for CIN 2 lasting 2 years.',
-          'Excisional treatment is preferred, and treatment with ablation is acceptable.',
-          'When considering ablative therapy, in particular cryotherapy, ablation is unacceptable in circumstances defined by the WHO.'
+          'Treatment is recommended for histologic HSIL (CIN 2) or histologic HSIL (unspecified) lasting 2 years.'
         }
-      }    
+      }
+  
     when ( // I3.4
       AgeInYears() >= 25 and AgeInYears() <= 50 and
-      (InitialCervicalHistologyInterpretedAsUnspecifiedHsilOrCin2InLast6Years is not null) and
-      Count(SubsequentLowGradeHistologyReportsSinceInitialUnspecifiedHsilOrCin2InLast6Years) = 5 and
-      Count(SubsequentLowGradeCytologyReportsSinceInitialUnspecifiedHsilOrCin2InLast6Years) >= 5 and
+      (InitialCervicalHistologyInterpretedAsUnspecifiedHsilOrCin2InLast31Years is not null) and
+      Count(FirstTwoNegativeHSILSurveillanceTests) >= 2 and
+      Count(NegativeSurveillanceTestDatesAfterHSILSurveillance) >= 3 and
       RecentlyRespondedYesToFuturePregnancyConcernsQuestion
```

Comment:
Sharon added a final line to this recommendation:

> AND the patient has only received ‘negative surveillance tests’ AFTER the above surveillance criteria


 as well as the following comment, which I think needs to be addressed here:

 

> "_Technical team - this is trying to say that as long as no abnormal results occurred after the initial intense surveillance period, the patient can continue to be screened every 3 years for 25 years._"

Reply:
@mrnosal are you planning to address this? This is the comment I was referring to: https://mitre.sharepoint.com/:w:/r/sites/CervicalCancerCDS/Shared%20Documents/CDS%20Development/L2%20Artifacts/Working%20Documents/CDC_CC_CDS_L2Development_v7.docx?d=wc90d0228d64f420bb1565fb44a712630&csf=1&web=1&e=6ocGpy&nav=eyJjIjo2ODEyNjYyNzh9.

The line that I linked above from the L2 does not seem to be reflected here in the L3. Is there a reason it was omitted, or should it be added?

Reply:
I'm addressing this in the HSIL branch for PR120

Reply:
This was addressed in PR #120:

https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/120/files#diff-79dcbe5d6961c9c3f6bda93de156d7f281b67a45deecf191662c1ac8cb1c9579.

I will resolve here.